src/vs/platform/terminal/common/terminal.ts

```diff
@@ -649,6 +650,11 @@ export interface ITerminalChildProcess {
 	 */
 	detach?(forcePersist?: boolean): Promise<void>;
 
+	/**
+	 * Frees the port and kills the process
+	 */
+	freePortKillProcess?(port: string): Promise<{ port: string; processId: string }>;
```

Comment:
Don't bother with that generic child process refactor I was talking about, we'll probably move all the core ones into a builtin extension eventually.

Reply:
yes agree, that's what I thought too